service/worker/addsearchattributes/workflow.go

```diff
@@ -152,6 +155,20 @@ func (a *activities) AddESMappingFieldActivity(ctx context.Context, params Workf
 	return nil
 }
 
+func (a *activities) isRetryableError(err error) bool {
+	var httpStatusCode int
+	if err, isElasticErr := err.(*elastic.Error); isElasticErr {
+		httpStatusCode = err.Status
+	}
```

Comment:
I think this is safer because it works with wrapped errors, and we return early. Leaving the status code variable as the zero value works because of the default branch, but I think it's better to just return early so that it's clear that this value has no meaning when it's not an ES error.

```go
var esErr *elastic.Error
if !errors.As(err, &esErr) {
  return false
}
httpStatusCode := esErr.Status
```